examples/adc.rs

```diff
@@ -0,0 +1,62 @@
+#![deny(unsafe_code)]
+#![no_main]
+#![no_std]
+#![feature(type_alias_enum_variants)]
+
+extern crate panic_itm;
+
+use cortex_m;
+use cortex_m_rt::entry;
+
+use stm32h7xx_hal::{
+    adc,
+    pac,
+    delay::Delay,
+    prelude::*,
+};
+
+use cortex_m_log::println;
+use cortex_m_log::{
+    destination::Itm, printer::itm::InterruptSync as InterruptSyncItm,
+};
+
+#[entry]
+fn main() -> ! {
+    let cp = cortex_m::Peripherals::take().unwrap();
+    let dp = pac::Peripherals::take().unwrap();
+    let mut log = InterruptSyncItm::new(Itm::new(cp.ITM));
+
+    // Constrain and Freeze power
+    println!(log, "Setup PWR...                  ");
+    let pwr = dp.PWR.constrain();
+    let vos = pwr.freeze();
+
+    // Constrain and Freeze clock
+    println!(log, "Setup RCC...                  ");
+    let rcc = dp.RCC.constrain();
+    // setting this per_ck to 4 mhz here as the last bit doesnt get sampled
+    // if we wouldn't do that
```

Comment:
The example should explain that the driver selects `adc_ker_ck` = `per_ck` [here](https://github.com/stm32-rs/stm32h7xx-hal/pull/12/files#diff-a4bc73d1077eb9a314216894b4643500R462). Therefore `per_ck` must be configured by the user. Setting `per_ck` to 4MHz selects `CSI` on the `per_ck` input multiplex.